EAC 573 S07 - Python coding in *COLAB*

Classes, Objects, Attributes and Methods

Classes
---
Object-Oriented Paradigm

Block of code

Reuse... instance

Inheritance

Attributes

Methods


CLASS

class ClassName:

  statements


[Caution: Indentation!]

In [1]:
# LPBProgram13.py
class MyClass:
  "My First Class"
  pi = 3.14
  def f(self):
    return("Hello World!")

In [2]:
x = MyClass.pi

In [3]:
print(x)

3.14


In [4]:
x = MyClass()

In [5]:
print(x.f())

Hello World!


In [6]:
print(x.pi)

3.14


In [7]:
#LBProgram14.py
class Employee:
  "Employee Class"  
  empCount = 0

  def __init__(self, name, salary):
    self.name = name
    self.salary = salary
    Employee.empCount += 1

  def displayCount(self):
    print("Total employee %d" %  Employee.empCount)

  def displayEmployee(self):
    print("Name: ", self.name, ", Salary: ", self.salary)
  

In [8]:
x = Employee("Barry", 100)
x.displayCount()

Total employee 1


In [9]:
y = Employee("Lara", 200)
y.displayCount()
x.displayCount()
print(Employee.empCount)

Total employee 2
Total employee 2
2


In [10]:
x.displayEmployee()
y.displayEmployee()
print(x.salary)

Name:  Barry , Salary:  100
Name:  Lara , Salary:  200
100


In [17]:
# CSV files
# upload  AccBatch.txt file to COLAB (under FILES)
import csv 
with open("AccBatch.txt", "rt") as f:
  reader = csv.reader(f)     # it will read all as string
  myList = list(reader)

In [18]:
myList

[['00001', ' 100.00', ' "Barry"', ' "1230001"'],
 ['00002', ' 200.00', ' "Lara"', ' "1230002"'],
 ['00003', ' 300.00', ' "Chris"', ' "1230003"'],
 ['00004', ' 400.00', ' "Luke"', ' "1230004"'],
 ['00005', ' 500.00', ' "Walton"', ' "1230005"']]

-Program-Exercise 21 (pe21.py)

Create a Class (with its attributes and methods) for a simple Bank System (use array or list for the accounts)

Class	 	 	 Account	

Attributes   Account	Number		 	 	 	 	   (str,	5	digits)
	 	 	 	     Account	Balance			 	 	 	 	 (float)
	 	 	 	 	   Customer	Name				 	 	 	 	 (str,	30	characters)
   	    	 	 Customer	TaxPayerNumber	     (str,	7	digits)

Methods		   Open	the	account	 	 	 	 	 		  OpenAccount	(init)
	 	 	 	 	   Get	the	current	balance	 	 		GetBalance	
	 	 	 	 	   Deposit	to	the	account	 	 	 	Deposit	
	 	 	 	 	   Withdraw	from	the	account			Withdraw	
	 	 	 	 	   Report	account	status	 	 	 		ReportStatus


-Program-Exercise 22 (pe22.py)

Create a program that uses the class Account (import from pe21.py) and create all accounts based on this file (“AccBatch.txt”)

"00001", 100.00, "Barry", "1230001"

"00002", 200.00, "Lara", "1230002"

"00003", 300.00, "Chris", "1230003"

"00004", 400.00, "Luke", "1230004"

"00005", 500.00, "Walton", "1230005" 


-Program-Exercise 23 (pe23.py)

Expand program 22 (keep it), now adding a solution to process all entries based on this file (“Acc0525.txt”)

"deposit", 00001, 1100.00

"deposit", 00003, 1300.00

"deposit", 00005, 1500.00

"withdraw", 00001, 400.00

"withdraw", 00002, 50.00

and report current balances of all accounts


-Program-Exercise 24 (pe24.py)

Expand program 23 (keep it), now adding a solution to process & write the Current Account Status to this file (“AccStatus0525.txt”)

Account Number, Account Balance, Customer Name and Customer TaxPayerNumber

